From 44b8e3b676746352cb3e08f76e58510c78209091 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Wed, 18 Sep 2024 14:10:58 +0200
Subject: [PATCH] tar/write.c: Support `sys/xattr.h`

Synchronize the last use of `attr/xattr.h` to support using
`sys/xattr.h` instead.  The former header is deprecated on GNU/Linux,
and this replacement makes it possible to build libarchive without
the `attr` package.
---
 tar/write.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tar/write.c b/tar/write.c
index 1d1139f4c..43cd70233 100644
--- a/tar/write.c
+++ b/tar/write.c
@@ -32,7 +32,9 @@
 #ifdef HAVE_SYS_STAT_H
 #include <sys/stat.h>
 #endif
-#ifdef HAVE_ATTR_XATTR_H
+#if HAVE_SYS_XATTR_H
+#include <sys/xattr.h>
+#elif HAVE_ATTR_XATTR_H
 #include <attr/xattr.h>
 #endif
 #ifdef HAVE_ERRNO_H
